runtime/src/mailbox.rs

```diff
@@ -57,23 +81,43 @@ impl Mailbox {
         }
     }
 
-    pub fn copy_to_mbox(&mut self, buf: &[Unalign<u32>]) {
+    pub fn flush(&mut self) {
+        let count = self.dlen_words();
+        let mbox = self.mbox.regs_mut();
+        for _ii in 0..count {
+            let _ = mbox.dataout().read();
+        }
+    }
+
+    pub fn copy_words_to_mbox(&mut self, buf: &[Unalign<u32>]) {
         let mbox = self.mbox.regs_mut();
         for word in buf {
             mbox.datain().write(|_| word.get());
         }
     }
 
+    pub fn copy_bytes_to_mbox(&mut self, buf: &[u8]) -> CaliptraResult<()> {
```

Comment:
To eliminate duplication, can write_response() call this function?

Reply:
Done.